doc/install/installation.md

```diff
@@ -0,0 +1,346 @@
+
+
+## 2. Ruby
+
+
+The use of Ruby version managers such as [RVM](http://rvm.io/), [rbenv](https://github.com/sstephenson/rbenv) or [chruby](https://github.com/postmodern/chruby) with Huginn in production frequently leads to hard-to-diagnose problems. Version managers are not supported and we strongly advise everyone to follow the instructions below to use a system Ruby.
+
+Remove the old Ruby versions if present:
+
+    sudo apt-get remove -y ruby1.8 ruby1.9
+
+Download Ruby and compile it:
+
+    mkdir /tmp/ruby && cd /tmp/ruby
+    curl -L --progress http://cache.ruby-lang.org/pub/ruby/2.2/ruby-2.2.2.tar.bz2 | tar xj
+    cd ruby-2.2.2
+    ./configure --disable-install-rdoc
+    make -j`nproc`
+    sudo make install
+
+Install the bundler and foreman gems:
+
+    sudo gem install bundler foreman --no-ri --no-rdoc
+
+## 3. System Users
+
+Create a user for Huginn:
+
+    sudo adduser --disabled-login --gecos 'Huginn' huginn
+
+## 4. Database
+
+Install the database packages
+
+    sudo apt-get install -y mysql-server mysql-client libmysqlclient-dev
+
+    # Pick a MySQL root password (can be anything), type it and press enter,
+    # retype the MySQL root password and press enter
+
+Check the installed MySQL version (remeber if its >= 5.5.3 for the `.env` configuration done later):
+
+    mysql --version
+
+Secure your installation
+
+    sudo mysql_secure_installation
+
+Login to MySQL
+
+    mysql -u root -p
+
+    # Type the MySQL root password
+
+Create a user for Huginn do not type the `mysql>`, this is part of the prompt. Change `$password` in the command below to a real password you pick
+
+    mysql> CREATE USER 'huginn'@'localhost' IDENTIFIED BY '$password';
+
+Ensure you can use the InnoDB engine which is necessary to support long indexes
+
+    mysql> SET storage_engine=INNODB;
+
+    # If this fails, check your MySQL config files (e.g. `/etc/mysql/*.cnf`, `/etc/mysql/conf.d/*`)
+    # for the setting "innodb = off"
+
+Grant the Huginn user necessary permissions on the database
+
+    mysql> GRANT SELECT, INSERT, UPDATE, DELETE, CREATE, DROP, INDEX, ALTER, LOCK TABLES ON `huginn_production`.* TO 'huginn'@'localhost';
+
+Quit the database session
+
+    mysql> \q
+
+Try connecting to the new database with the new user
+
+    sudo -u huginn -H mysql -u huginn -p -D huginn_production
+
+    # Type the password you replaced $password with earlier
+
+You should now see `ERROR 1049 (42000): Unknown database 'huginn_production'` which is fine because we will create the database later.
+
+You are done installing the database and can go back to the rest of the installation.
+
+
+## 6. Huginn
+
+### Clone the Source
+
+    # We'll install Huginn into home directory of the user "huginn"
+    cd /home/huginn
```

Comment:
The idea is to not switch users at all during the guide, you start with either root or a sudo user (which I probably should mention somewhere) and never have to worry about which user has to execute which command.